services/indexeddbService.ts

```diff
@@ -94,19 +94,25 @@ export async function insertSiteFeed(siteFeed: SiteFeed) {
 
 // siteFeed must contain entire class, both old and new visited sites !!
 export async function updateSiteFeed(siteFeed: SiteFeed) {
-  const request = indexedDB.open(databaseName, databaseVersion);
+  return new Promise<void>((resolve, reject) => {
```

Comment:
https://stackoverflow.com/questions/37104199/how-to-await-for-a-callback-to-return